core/aws-core/src/main/java/software/amazon/awssdk/awscore/AwsRequest.java

```diff
@@ -107,6 +119,17 @@ public Builder overrideConfiguration(Consumer<AwsRequestOverrideConfiguration.Bu
             return this;
         }
 
+        @Override
+        public List<SdkPlugin> registeredPlugins() {
+            return registeredPlugins;
+        }
+
+        @Override
+        public Builder addPlugin(SdkPlugin plugin) {
```

Comment:
shouldn't AwsRequest.Builder interface re-define the method in the interface `public Builder addPlugin(SdkPlugin plugin)`, so that the return type is AwsRequest.Builder and not SdkRequest.Builder?